dlt/destinations/impl/databricks/databricks.py

```diff
@@ -0,0 +1,336 @@
+
+        # extract and prepare some vars
+        bucket_path = (
+            NewReferenceJob.resolve_reference(file_path)
+            if NewReferenceJob.is_reference_job(file_path)
+            else ""
+        )
+        file_name = (
+            FileStorage.get_file_name_from_file_path(bucket_path) if bucket_path else file_name
+        )
+        from_clause = ""
+        credentials_clause = ""
+        files_clause = ""
+        # stage_file_path = ""
+        format_options = ""
+        copy_options = "COPY_OPTIONS ('mergeSchema'='true')"
+
+        if bucket_path:
+            bucket_url = urlparse(bucket_path)
+            bucket_scheme = bucket_url.scheme
+            # referencing an external s3/azure stage does not require explicit credentials
+            if bucket_scheme in ["s3", "az", "abfs", "gc", "gcs"] and stage_name:
+                from_clause = f"FROM ('{bucket_path}')"
+            # referencing an staged files via a bucket URL requires explicit AWS credentials
+            elif (
+                bucket_scheme == "s3"
+                and staging_credentials
+                and isinstance(staging_credentials, AwsCredentialsWithoutDefaults)
+            ):
+                s3_creds = staging_credentials.to_session_credentials()
+                credentials_clause = f"""WITH(CREDENTIAL(
+                AWS_ACCESS_KEY='{s3_creds["aws_access_key_id"]}',
+                AWS_SECRET_KEY='{s3_creds["aws_secret_access_key"]}',
+                AWS_SESSION_TOKEN='{s3_creds["aws_session_token"]}'
+                ))
+                """
+                from_clause = f"FROM '{bucket_path}'"
+            elif (
+                bucket_scheme in ["az", "abfs"]
+                and staging_credentials
+                and isinstance(staging_credentials, AzureCredentialsWithoutDefaults)
+            ):
+                # Explicit azure credentials are needed to load from bucket without a named stage
+                credentials_clause = f"""WITH(CREDENTIAL(AZURE_SAS_TOKEN='{staging_credentials.azure_storage_sas_token}'))"""
+                # Converts an az://<container_name>/<path> to abfss://<container_name>@<storage_account_name>.dfs.core.windows.net/<path>
+                # as required by snowflake
+                _path = bucket_url.path
+                bucket_path = urlunparse(
+                    bucket_url._replace(
+                        scheme="abfss",
+                        netloc=f"{bucket_url.netloc}@{staging_credentials.azure_storage_account_name}.dfs.core.windows.net",
+                        path=_path,
+                    )
+                )
+                from_clause = f"FROM '{bucket_path}'"
+            else:
+                raise LoadJobTerminalException(
+                    file_path,
+                    f"Databricks cannot load data from staging bucket {bucket_path}. Only s3 and azure buckets are supported",
+                )
+        else:
+            raise LoadJobTerminalException(
```

Comment:
somehow i think it would be nice to be able to define in the capabilities which staging filesystem types are supported by each destination. not sure wether to add this now or later, but it would save some of this code in a few places and we could also catch config errors earlier.

Reply:
Good idea to have this in capabilities. Can do that and refactor in another PR.